File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    width = 3
    length = 3
    sum_by = brown + yellow
    # mul_by = brown * yellow
    
    for i in range(1, sum_by):
        if(sum_by % i == 0):
            width = max(i, sum_by//i)
            length = min(i, sum_by//i)
            if((width - 2) * (length - 2) == yellow):
                break
    
    answer = [width, length]
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int total = brown + yellow; // 전체 격자 수

        // 세로는 최소 3부터 시작 (가로 >= 세로)
        for (int height = 3; height <= Math.sqrt(total); height++) {
            if (total % height == 0) {
                int width = total / height;
                if ((width - 2) * (height - 2) == yellow) {
                    answer[0] = width;
                    answer[1] = height;
                    break;
                }
            }
        }
        return answer;
    }
}